client/cspaillier.go

```diff
@@ -6,46 +6,70 @@ import (
 	"math/big"
 )
 
-func (c *Client) CSPaillier(pubKeyPath string, m, label *big.Int) (bool, error) {
+type CSPaillierClient struct {
+	genericClient
+	encryptor *encryption.CSPaillier
+	label, m  *big.Int
+}
+
+func NewCSPaillierClient(endpoint string, pubKeyPath string, m, l *big.Int) (*CSPaillierClient, error) {
+	genericClient, err := NewGenericClient(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
 	encryptor, err := encryption.NewCSPaillierFromPubKeyFile(pubKeyPath)
 	if err != nil {
-		return false, err
+		return nil, err
 	}
 
-	c.handler.paillierEncryptor = encryptor
-	u, e, v, _ := c.handler.paillierEncryptor.Encrypt(m, label)
+	return &CSPaillierClient{
+		genericClient: *genericClient,
+		encryptor:     encryptor,
+		m:             m,
+		label:         l,
+	}, nil
+}
 
-	if err = c.openCSPaillier(m, u, e, v, label); err != nil {
-		return false, err
+func (c *CSPaillierClient) Run() error {
```

Comment:
Done. All exported functions in client package are now (briefly) documented - this includes initialization functions and `Run()` functions.

I intentionally didn't document what all the parameters for clients' initialization functions are. I will ask @miha-stopar to help me with this some time in the future to avoid silly mistakes.